src/hardware/vga_crtc.cpp

```diff
@@ -189,21 +195,21 @@ void vga_write_p3d5(io_port_t, io_val_t value, io_width_t)
 
 		vga.crtc.maximum_scan_line.data = new_val.data;
 
-		if (vga.draw.is_double_scanning) {
-			auto flipped = new_val;
-			flipped.start_vertical_blanking_bit9.flip();
-
-			// Only true if only `start_vertical_blanking_bit9` has
-			// been changed and all other bits are unchanged.
-			if (old_val.data == flipped.data) {
-				VGA_StartResize();
-			}
-
+		if (vga.draw.is_double_scanning && !is_svga_or_vesa(vga.mode)) {
+			// This is a potential fast path where we can skip VGA resizing.
+			// A rare few games (ex. Wing Commander 3) fiddle with the double-scan bit
+			// in SVGA/VESA modes.
```

Comment:
I'd prefer if you forced a resizing as I don't like it if this algorithm becomes too "distributed". We don't want a "fast" path here as the resize logic is fast enough, but we absolutely want as much clarity as we can achieve.

What should happen here is detecting the conditions that require a resize, then modify the big resize function (if needed; maybe triggering a resize here is what's missing).

Reply:
"Fast path" is probably the wrong wording as it's not really about performance.  My first thought was just to call `VGA_StartResize` all the time but you tried that and it apparently broke Legend of Kyriandia then #3279 restored this behavior.

I guess I'm unclear what you mean if not that.

Reply:
Okay, just proceed as you think is best then we'll talk at review time 😄

Reply:
I think I see what you mean now.  I did some regression testing and I think we can get rid of this special double-scan path.

If you avoid all checking, the Legend of Kyrandia bug returns.  However, if you keep only the bottom (general) path, everything seems to work.

I force pushed this to a single commit to remove the top branch.  It is nice when you can fix a bug simply by deleting code :smile:

Reply:
Yeah, this looks good. I meant only to trigger the resize from here, then updating the VGA state based on the register value change should be done in the big `setup_drawing()` function in `vga_draw.cpp` (see below).

If this simplification works, then great 🎉 I will need to read `setup_drawing()` again in light of this change for correctness, then test with a bunch of special VGA programs that do weird things to the hardware... The good news is the outcome of that will be a VGA regression testing wiki page that we sorely need.

Great job so far @weirddan455 👍🏻 Good to see I'm not the only one hacking away at the VGA code.

```cpp
// Can change the following VGA state:
//
//   vga.draw.address_line_total
//   vga.draw.blocks
//   vga.draw.delay.hdend
//   vga.draw.dos_refresh_hz
//   vga.draw.line_length
//   vga.draw.linear_base
//   vga.draw.linear_mask
//   vga.draw.mode
//   vga.draw.resizing
//   vga.draw.uses_vga_palette
//   vga.draw.vblank_skip
//   vga.draw.vret_triggered
//
ImageInfo setup_drawing()
{
```

Reply:
`vga.draw.address_line_total` (only VGA variable that previously got set here) already gets re-set in `setup_drawing()` according to the register values (and the code there correctly handles edge cases unlike here).  I don't think anything should need changing there.

It seems this "fast path" logic existed before your improvements to double-scanning and so is incorrect now.  Hopefully there won't be any problems just removing it as that would be the simplest solution.

Reply:
> It seems this "fast path" logic existed before your improvements to double-scanning and so is incorrect now.

Yeah, and I didn't like it, just was reluctant to touch it 😄 

> vga.draw.address_line_total (only VGA variable that previously got set here) already gets re-set in setup_drawing() according to the register values (and the code there correctly handles edge cases unlike here). I don't think anything should need changing there.

That's my hope. I'll just need to test the crap out of it 😄